### PR TITLE
Display 'No matching feature flags found' message when no flags found from search

### PR DESCRIPTION
**Pull request by zackcl**
This PR displays the message "No matching feature flags found" on the Feature Flags Root Section Card Table instead of "No feature flags are defined. Create a new feature flag." when no matching feature flags are found from the search.

Screenshot 2024-09-06 at 2 02 22 PM

### CODE REVIEW

**Comment by zackcl** [comment]
@Yagnik56 @VivekFitkariwala Please review and approve this PR.

**Comment by zackcl** [comment]
OK, I will first resolve conflicts.

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts [hunk 1]
```diff
@@ -91,6 +92,7 @@ export class FeatureFlagRootSectionCardComponent {
   onSearch(params: CommonSearchWidgetSearchParams<FLAG_SEARCH_KEY>) {
     this.featureFlagService.setSearchString(params.searchString);
     this.featureFlagService.setSearchKey(params.searchKey as FLAG_SEARCH_KEY);
+    this.isSearchActive$ = of(!!params.searchString);
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts:95]: This is not needed

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts [hunk 2]
```diff
@@ -49,6 +49,7 @@ export class FeatureFlagRootSectionCardComponent {
   searchKey$ = this.featureFlagService.searchKey$;
   searchParams$ = this.featureFlagService.searchParams$;
   selectRootTableState$ = this.featureFlagService.selectRootTableState$;
+  isSearchActive$ = of(false);
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts:52]: You should subscribe the value from the store.

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts (hunk 1 above)
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts:95]: This is not needed

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ import { FLAG_SEARCH_KEY, IMenuButtonItem } from 'upgrade_types';
 import { RouterModule } from '@angular/router';
 import { MatTableDataSource } from '@angular/material/table';
 import { DialogService } from '../../../../../../../shared/services/common-dialog.service';
-import { Observable, Subscription, map } from 'rxjs';
+import { Observable, Subscription, map, of } from 'rxjs';
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-section-card/feature-flag-root-section-card.component.ts:16]: Remove Subscription

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]

**Comment by zackcl** [comment]
@VivekFitkariwala I resolved all your comments. Please feel free to approve this PR.

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**